arcsi/api/show.py

```diff
@@ -282,13 +297,45 @@ def view_show(id):
     else:
         return make_response("Show not found", 404, headers)
 
+# We use this route on the legacy front-end show page
 @arcsi.route("show/<string:show_slug>/archive", methods=["GET"])
 def view_show_archive(show_slug):
+    do = DoArchive()
+    show_query = Show.query.filter_by(archive_lahmastore_base_url=show_slug)
+    show = show_query.first_or_404()
+    if show:
+        show_json = show_schema.dump(show)
+        show_items = [
+            show_item
+            for show_item in show_json["items"]
+            if datetime.strptime(show_item.get("play_date"), "%Y-%m-%d")
+            + timedelta(days=1)
+            < datetime.today()
+        ]
+        for show_item in show_items:
+            show_item["image_url"] = do.download(
+                show.archive_lahmastore_base_url, show_item["image_url"]
+            )
+        return json.dumps(show_items)
+    else:
+        return make_response("Show not found", 404, headers)
+    #show = show_query.first()
+    #if show:
+    #    show_items = show.items.filter(Item.play_date < datetime.today() - timedelta(days=1)).all()
+    #    return items_schema.dump(show_items)
+    #else:
+    #    return make_response("Show episodes not found", 404, headers)
+
+# This will be the one that we are gonna use at the new page 
+@arcsi.route("show/<string:show_slug>/page", methods=["GET"])
+def view_show_page(show_slug):
     show_query = Show.query.filter_by(archive_lahmastore_base_url=show_slug)
     show = show_query.first()
     if show:
-        show_items = show.items.filter(Item.play_date < datetime.today() - timedelta(days=1)).all()
-        return items_basic_schema.dump(show)
+        # subquery = session.query(Item.id).filter(blabla -timedelta(day=1)).all().subquery()
+        # query = session.query(Show).filter_by(blabla).(Item.id.in_(subquery))
```

Comment:
Would you like to finish this in this branch or in another later?

Reply:
Yes, I'll give it a try with the next commit.